packages/ui/src/services/breakpoints/plugin/create-service.ts

```diff
@@ -0,0 +1,123 @@
+import { App, Ref, computed, watch, ComputedRef } from 'vue'
+
+import { useDocument, useWindowSize } from '../../../composables'
+
+import { isClient } from '../../../utils/ssr-utils'
+import { warn, generateUniqueId } from '../../utils'
+import { getGlobalProperty } from '../../../vuestic-plugin/utils'
+import { addOrUpdateStyleElement } from '../../dom-functions'
+
+import { GlobalConfig } from '../../global-config/types'
+import { ThresholdsKey, BreakpointsConfig, BodyClass } from '../types'
+
+export const createBreakpointsConfigPlugin = (app: App) => {
+  const globalConfig: Ref<GlobalConfig> | undefined = getGlobalProperty(app, '$vaConfig')?.globalConfig
+  if (!globalConfig) {
+    warn('createBreakpointsConfigPlugin: globalConfig is not defined!')
+    return {}
+  }
+
+  const breakpointsConfig: ComputedRef<BreakpointsConfig> = computed(() => {
+    const breakpoints = globalConfig.value.breakpoints
+    if (!breakpoints) { warn('createBreakpointsConfigPlugin: breakpointsConfig is not defined!') }
+    return breakpoints ?? {} as BreakpointsConfig
+  })
+
+  if (!breakpointsConfig.value.enabled) {
+    return {}
+  }
+
+  if (!breakpointsConfig.value.thresholds || !Object.values(breakpointsConfig.value.thresholds).length) {
+    warn('createBreakpointsConfigPlugin: there are no defined thresholds!')
+    return {}
+  }
+
+  const { windowSizes } = useWindowSize()
+
+  const isMounted = computed(isClient)
+  const currentBreakpoint = computed(() => {
+    if (!isMounted.value || !windowSizes.width) { return }
+
+    return Object.entries(breakpointsConfig.value.thresholds).reduce((acc: string, [key, value]) => {
+      if (windowSizes.width! >= value) { acc = key }
+      return acc
+    }, 'xs') as unknown as ThresholdsKey
+  })
+
+  const screenClasses = computed(() =>
+    (Object.keys(breakpointsConfig.value.thresholds) as ThresholdsKey[])
+      .reduce((acc: Record<ThresholdsKey, BodyClass>, threshold: ThresholdsKey) => {
+        acc[threshold] = `va-screen-${threshold}`
+        return acc
+      }, {} as Record<ThresholdsKey, BodyClass>))
+
+  const generateHelpersMediaCss = () => {
+    let result = ''
+
+    Object.values(breakpointsConfig.value.thresholds)
+      .forEach((thresholdValue, index) => {
+        result += `@media screen and (min-width: ${thresholdValue}px) {`
+        // 0.2 coefficient for xs threshold and 1 for xl, experimental value for now
```

Comment:
this is not entirely true: what if the user defined the next threshold:
```ts
{
  xs: 640,
  lg: 1024,
  xl: 1920,
}
```

Reply:
He must. Or fill `0` for thresholds he doesn't need.

<img width="904" alt="image" src="https://user-images.githubusercontent.com/64714442/190148693-ef04769e-b2b9-4081-ba18-7cd3a6cb552b.png">